### PR TITLE
ValidatorStorage update and SqliteMemoryValidatorStorage Impl.

### PR DESCRIPTION
**Pull request by Sid-Data-Universe**

### CODE REVIEW

File: docs/validator.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ As of Dec 11th 2023, the expected cost number of DataItems queried via Apify is
 
 # System Requirements
 
-Validators do not require a GPU and should be able to run on a relatively low-tier machine, as long as it has sufficient network bandwidth and disk space.
+Validators require at least 32 GB of RAM but do not require a GPU and should be able to run on a relatively low-tier machine otherwise, as long as it has sufficient network bandwidth and disk space.
```
Comment (RusticLuftig) [inline docs/validator.md:17]: I think we should recommend a CPU with a few cores and remove the disk space note.

File: neurons/validator.py [hunk 2]
```diff
@@ -172,10 +174,7 @@ async def _update_and_get_miner_index(
 
     def _get_miner_index(self, hotkey: str) -> Optional[ScorableMinerIndex]:
```
Comment (RusticLuftig) [inline neurons/validator.py:175]: Is it worth removing this helper method now?

**Review by RusticLuftig — APPROVED** [review]

File: neurons/validator.py (hunk 2 above)
Comment (Sid-Data-Universe) [reply neurons/validator.py]: Removed. Good catch.

File: docs/validator.md (hunk 1 above)
Comment (Sid-Data-Universe) [reply docs/validator.md]: Updated.

**Review by RusticLuftig — APPROVED** [review]

**Pull request merged**